### PR TITLE
librados: fix leak in getxattr and getxattrs

### PR DESCRIPTION
**Pull request by aclamk**
Issue was related to missing put() on AioCompletionImpl.

### CODE REVIEW

**Comment by aclamk** [comment]
This fixed issue that can be noticed with:
valgrind --leak-check=full ./bin/ceph_test_rados_api_aio --gtest_filter=*XattrsRoundTrip*

**Comment by aclamk** [comment]
Cleared DNM, fixed typo.

File: src/librados/librados_c.cc [hunk 1]
```diff
@@ -2597,6 +2600,7 @@ static void rados_aio_getxattrs_complete(rados_completion_t c, void *arg) {
     cdata->it = 0;
     cdata->user_completion.finish(0);
   }
+  cdata->operational_completion->put();
```
Comment (tchaikov) [inline src/librados/librados_c.cc]: @aclamk Adam, i think it's not an intuitive fix. probably an alternative is to set the refcount of `AioCompletionImpl` to 0 when it is created. as we always increase its refcount right after capturing its strong reference. in this case, `C_AioCompleteAndSafe` calls `c->get()` when assigning `cc` to `this->c`.

Comment (aclamk) [reply src/librados/librados_c.cc]: @tchaikov Thanks! I did not understand that context passed to completion is the one used in invocation.

File: src/librados/librados_c.cc [hunk 2]
```diff
@@ -2545,6 +2545,7 @@ static void rados_aio_getxattr_complete(rados_completion_t c, void *arg) {
   }
   cdata->user_completion.finish(rc);
   delete cdata;
+  ((librados::AioCompletionImpl*)c)->put();
```
Comment (tchaikov) [inline src/librados/librados_c.cc]: nit, can we follow the same patter of 

https://github.com/ceph/ceph/blob/c28876f0084d583de0550b105c2ae45222fd30e8/src/librados/IoCtxImpl.cc#L1178-L1179

i.e. call `put()` before `delete cdata`? it's not big deal, but just for the sake of consistency.

Comment (aclamk) [reply src/librados/librados_c.cc]: @tchaikov Synced.

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-12-21_03:01:01-rados-wip-kefu-testing-2019-12-21-0133-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-12-23_16:20:31-rados-wip-kefu-testing-2019-12-21-0133-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/43396
- https://tracker.ceph.com/issues/43415
- https://tracker.ceph.com/issues/43414
- https://tracker.ceph.com/issues/43416

**Pull request merged**